src/commands/mod.rs

```diff
@@ -1841,6 +1844,57 @@ fn edit_description(
     Ok(text_util::complete_newline(description))
 }
 
+fn edit_sparse(
+    cwd: &Path,
+    workspace_root: &Path,
+    repo_path: &Path,
+    sparse: &[RepoPath],
+    settings: &UserSettings,
+) -> Result<Vec<RepoPath>, CommandError> {
+    let file_path = (|| -> Result<_, io::Error> {
+        let mut file = tempfile::Builder::new()
+            .prefix("editor-")
+            .suffix(".jjsparse")
+            .tempfile_in(repo_path)?;
+        for sparse_path in sparse {
+            file.write_all(
+                file_util::relative_path(cwd, &sparse_path.to_fs_path(workspace_root))
+                    .to_str()
+                    .unwrap()
+                    .as_bytes(),
+            )?;
+            file.write_all(b"\n")?;
+        }
+        file.write_all(b"\nJJ: Lines starting with \"JJ: \" (like this one) will be removed.\n")?;
+        let (_, path) = file.keep().map_err(|e| e.error)?;
+        Ok(path)
+    })()
+    .map_err(|e| {
+        user_error(format!(
+            r#"Failed to create sparse patterns file in "{path}": {e}"#,
+            path = repo_path.display()
+        ))
+    })?;
+
+    run_ui_editor(settings, &file_path)?;
+
+    let sparse_patterns_string = fs::read_to_string(&file_path).map_err(|e| {
+        user_error(format!(
+            r#"Failed to read sparse patterns file "{path}": {e}"#,
+            path = file_path.display()
+        ))
+    })?;
+    // Delete the file only if everything went well.
+    // TODO: Tell the user the name of the file we left behind.
```

Comment:
Ah, okay. Removed.

Reply:
And I used `BufReader::new().lines()` instead of `read_to_string()` in case the file has a lot of lines (unlikely, but if we're really dealing with big repos where `sparse` is necessary, I reckon it's possible this file might have a lot of patterns).

Reply:
We read the whole file into memory so we can create a `Matcher` object from it whenever we snapshot the working copy anyway, so I'm not worried about reading it into a string here, but I'm also fine with reading it line by line.

Reply:
I was gonna change this back to using `read_to_string()` to simplify the code, but I actually feel like `BufReader::new().lines()` is about as readable so I'll keep it this way.